planner/core/task.go

```diff
@@ -818,6 +818,19 @@ func (t *rootTask) MemoryUsage() (sum int64) {
 	return sum
 }
 
+// attach2Task attach limit to different cases.
+// For Normal Index Lookup
+// 1: attach the limit to table side or index side of normal index lookup cop task. (normal case, old code, no more
+// explanation here)
+//
+// For Index Merge:
+// 2: attach the limit to **table** side for index merge intersection case, cause intersection will invalidate the
+// fetched limit+offset rows from each partial index plan, you can not decide how many you want in advance, actually.
+//
+// 3: attach the limit to **index** side for index merge union case, because each index plan will output the fetched
```

Comment:
Another precondition is that there is no table filter?

Reply:
yes, the entry point is `cop.indexPlanFinished` = true, after we building from index merge read, if we find table side is not a pure table scan, this flag `cop.indexPlanFinished` will be turned as true.
```
	if prop.TaskTp == property.RootTaskType {
		cop.indexPlanFinished = true
		task = cop.convertToRootTask(ds.SCtx())
	} else {
		_, pureTableScan := ts.(*PhysicalTableScan)
		if !pureTableScan {
			cop.indexPlanFinished = true
		}
		task = cop
	}
```
I will make it more clear